src/PrestaShopBundle/Utils/ImmutableFloat.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace PrestaShopBundle\Utils;
+
+/**
+ * Immutable value-object that contains a float value.
+ */
+class ImmutableFloat
+{
+    /**
+     * @var float
+     */
+    private $value;
+
+    /**
+     * @param float $value
+     *
+     * @throws \InvalidArgumentException if the passed value is not a valid float
+     */
+    public function __construct($value)
+    {
+        if (!is_float($value)) {
+            throw new \InvalidArgumentException('Invalid argument: "%s" is not a valid float', $value);
+        }
+
+        $this->value = $value;
+    }
+
+    /**
+     * Constructs a new instance from a string value.
+     * It supports dot and comma-separated values.
+     *
+     * @param string $value
+     *
+     * @return ImmutableFloat
+     */
+    public static function fromString($value)
+    {
+        $value = (string) $value;
+        if ($value === '') {
+            return new self(0.0);
```

Comment:
As the constructor is public and the class is not `final` IMHO you should use `static` instead of `self`